portal-impl/src/com/liferay/portal/tools/deploy/BaseDeployer.java

```diff
@@ -116,9 +116,15 @@ else if (fileName.endsWith(".jar")) {
 			}
 		}
 
-		try (BaseDeployer temp = new BaseDeployer(wars, jars)) {
+		BaseDeployer baseDeployer = new BaseDeployer(wars, jars);
+
+		try {
+			baseDeployer.close();
 		}
 		catch (IOException ioe) {
+			if (_log.isWarnEnabled()) {
```

Comment:
so for every replacement, there should be a log? Because some replacement is adding "throw IOException" to method signature right now, which is not a "quiet" cleanUp

Reply:
Yes, I think so.